all.sas

```diff
@@ -1748,33 +1748,43 @@ Usage:
       if debug ge '"131"' then put '>>weboutEND<<';
     run;
 
-    %if %symexist(_metaport) %then %do;
+    %if "&sysprocessmode " = "SAS Stored Process Server " %then %do;
       data _null_;
-        if symexist('sysprocessmode') then
-          if symget("sysprocessmode")="SAS Stored Process Server" then do;
-            rc=stpsrvset('program error', 0);
-            call symputx("syscc",0,"g");
-          end;
+        putlog 'stpsrvset program error and syscc';
+        rc=stpsrvset('program error', 0);
+        call symputx("syscc",0,"g");
       run;
     %end;
+
+    %if "%substr(&sysvlong.xxxxxxx,1,9)" ne "9.04.01M3" %then %do;
+      %put NOTE: Ending SAS session due to:;
+      %put NOTE- &msg;
+      endsas;
+    %end;
+
     /**
-      * endsas is reliable but kills some deployments.
+      * endsas is reliable but kills 9.4m3 deployments by hanging multibridges.
       * Abort variants are ungraceful (non zero return code)
       * This approach lets SAS run silently until the end :-)
+      * Caution - fails when called within a %include within a macro
```

Comment:
I feel like this deserves mentioning in the header :)

Reply:
done